### DISCUSSION TITLE
Creating a Custom Spoke Client

### DISCUSSION THREAD

**Discussion by daCking15** [mozilla/hubs#3327] [Spoke Questions] [answered]
Steps I have taken

1. Created a Hubs-Cloud Personal instance
2. Cloned the Mozilla Spoke repository
3. Added deploy.js script from [Pull Request](https://github.com/MozillaReality/ieeevr-spoke/pull/1)
4. Ran npm run login on the hubs repository and copied the .ret.credentials file into the spoke repository.
5. Changed the package.json file to include the deploy script.
6. Added the necessary yarn modules.
7. Changed the .env.prod file to match the values given by jamesckane on Discord:
![image](https://user-images.githubusercontent.com/15696956/98986016-57e14a00-24f2-11eb-9c0b-cd25de05e98c.png)

Result

I am able to deploy my custom spoke respository using yarn run deploy, but when I try to create a new project, I am getting the following errors:
![image](https://user-images.githubusercontent.com/15696956/98985950-40a25c80-24f2-11eb-884f-639cdbf10c7e.png)

Additional Information

- I am getting these errors as well, but after checking my other Hubs instance, it appears that they show up there to and do not stop Spoke from working properly:
![image](https://user-images.githubusercontent.com/15696956/98986427-e655cb80-24f2-11eb-85e3-9b1e829b276e.png)

- jamesckane claims to have gotten his custom spoke client running using the .env.prod values referenced in step 7.

**Comment by daCking15** [comment, marked as answer]
I figured it out, it was due to yarn package inconsistencies. I ran yarn upgrade and I think that is what messed things up. After doing a clean install of yarn, the above steps worked and I no longer received the errors. However, the "refused to prefetech content" errors still show up even though spoke runs as expected.

**Reply by daCking15** [reply]
The prefetch errors have stopped showing up too, must have been fixed by Mozilla somehow.

**Comment by angelcabrera-nr** [comment]
Thanks @daCking15, thanks to this post I was able to deploy my Custom Spoke Client without problems.

For future people who wonder where they can get all the information that goes in the environment variables, they can be found in the `Outputs` tab of your Cloudformation stack

![image](https://user-images.githubusercontent.com/73490994/199277049-ff8506d4-33c7-49a3-bb29-83f26f3ba52b.png)
